torch/nn/utils/clip_grad.py

```diff
@@ -29,17 +29,17 @@ def clip_grad_norm_(
     """
     if isinstance(parameters, torch.Tensor):
         parameters = [parameters]
-    parameters = [p for p in parameters if p.grad is not None]
+    grads = [p.grad for p in parameters if p.grad is not None]
```

Comment:
@albanD the lint failures where cases where because `mypy` now has typing information for `.grad` and needs to be convinced that the value won't be `None`.

Reply:
Ho and `mypy` doesn't see the `is not None` check?

Reply:
I think the type refinement maybe only works within a single scope, so once you leave the list comprehension it forgets that `p.grad` won't be none. However this new comprehension results in `grads` being non-optional Tensors so it does work.